src/ncp/ncp_base_mtd.cpp

```diff
@@ -1228,17 +1228,25 @@ otError NcpBase::EncodeOperationalDataset(const otOperationalDataset &aDataset)
 
     if (aDataset.mComponents.mIsActiveTimestampPresent)
     {
+        const otTimestamp &activeTimestamp = aDataset.mActiveTimestamp;
+
         SuccessOrExit(error = mEncoder.OpenStruct());
         SuccessOrExit(error = mEncoder.WriteUintPacked(SPINEL_PROP_DATASET_ACTIVE_TIMESTAMP));
-        SuccessOrExit(error = mEncoder.WriteUint64(aDataset.mActiveTimestamp));
+        SuccessOrExit(error = mEncoder.WriteUint64((activeTimestamp.mSeconds << 16) |
```

Comment:
nit: defininig otTimestamp as an union allows accesing the uint64_t value directly.

Reply:
We can't do that, there is no guarantee of bit fields order. See https://stackoverflow.com/questions/19376426/order-of-fields-when-using-a-bit-field-in-c.